django/applications/catmaid/control/annotation.py

```diff
@@ -1349,15 +1347,10 @@ def list_annotations(request:HttpRequest, project_id=None) -> JsonResponse:
 
     # Create a set mapping annotation names to its users
     ids = {}
-    annotation_dict = {} # type: Dict
+    annotation_dict:DefaultDict[Any, List[Dict]] = defaultdict(list)
     for annotation, aid, uid, username in annotation_tuples:
         ids[aid] = annotation
-        ls = annotation_dict.get(aid)
-        if ls is None:
-            ls = []
-            annotation_dict[aid] = ls
-        if uid is not None:
-            ls.append({'id': uid, 'name': username})
+        annotation_dict[aid].append({'id': uid, 'name': username})
```

Comment:
Ah, right. Didn't stare at this one quite hard enough.

Reply:
Having this as a defaultdict will subtly change the semantics in a way that will hopefully not matter (entries will only be made if uid is not None). Making the adjustment now, but if it's safer to go back to Dict and the old logic for that, that's fine too.

Reply:
I think you may be right here - having the key with an empty list may be necessary somewhere downstream, where the defaultdict would not add the key unless the value was operated. I don't know off the top of my head how/where this endpoint is used but it may be best to go back to the previous implementation for that reason.

Reply:
Revised it back (but hopefully left it easier to read)

Reply:
LGTM